src/main/java/dobby/command/FindtypeCommand.java

```diff
@@ -0,0 +1,41 @@
+package dobby.command;
+
+import java.util.List;
+
+import dobby.DobbyException;
+import dobby.TaskList;
+
+public class FindtypeCommand implements Command {
+
+    protected static final String USAGE = "findtype _T/D/E_";
+
+    @Override
+    public String parseInput(TaskList tasks, String text) throws DobbyException {
+        String message = "";
+        assert text.startsWith("findtype") : "Findtype command must start with findtype";
+        try {
+            String type = text.substring(text.indexOf(' ') + 1);
+
+            if (type.length() > 1) {
+                throw new DobbyException("Incorrect usage of command. "
+                        + "Please try again.\n  " + USAGE);
+            }
+
+            type = type.toUpperCase();
+            if (!(type.equals("T") || type.equals("D") || type.equals("E"))) {
```

Comment:
you might want to abstract this into a boolean variable, like boolean isNotTaskType = !(type.equals("T") || type.equals("D") || type.equals("E");